core/sup_tree_core/periodic_log/writer.ex

```diff
@@ -31,53 +29,32 @@ defmodule AntikytheraCore.PeriodicLog.Writer do
   @impl true
   def init([build_mod, file_name | opts]) do
     write_to_terminal? = Keyword.get(opts, :write_to_terminal, false)
-    handle = FileHandle.open(AntikytheraCore.Path.core_log_file_path(file_name), write_to_terminal: write_to_terminal?)
-    timer = arrange_next_rotation(nil)
+    log_file_path = AntikytheraCore.Path.core_log_file_path(file_name)
+    log_state = LogRotation.init(@rotate_interval, log_file_path, write_to_terminal: write_to_terminal?)
     build_state = build_mod.init()
-    {:ok, %State{file_handle: handle, empty?: true, timer: timer, build_mod: build_mod, build_state: build_state}, @interval}
+    {:ok, %State{log_state: log_state, build_mod: build_mod, build_state: build_state}, @interval}
   end
 
   @impl true
-  def handle_info(:timeout, %State{file_handle: handle, timer: timer, build_mod: build_mod, build_state: build_state} = state) do
-    {message, next_state} = build_mod.build_log(build_state)
-    if message != nil do
-      msg = {Time.now(), :info, ContextId.system_context(), message}
-      case FileHandle.write(handle, msg) do
-        {:kept_open, new_handle} -> {:noreply, %State{state | file_handle: new_handle, empty?: false, build_state: next_state}, @interval}
-        {:rotated  , new_handle} ->
-          # Log file is just rotated as its size has exceeded the upper limit.
-          # Note that the current message is written to the newly-opened log file and thus it's not empty.
-          new_timer = arrange_next_rotation(timer)
-          {:noreply, %State{state | file_handle: new_handle, empty?: false, timer: new_timer, build_state: next_state}, @interval}
+  def handle_info(:timeout, %State{log_state: log_state, build_mod: build_mod, build_state: build_state} = state) do
+    {message, next_build_state} = build_mod.build_log(build_state)
+    next_state =
+      if message != nil do
```

Comment:
I think the following is better.
The reasons are the following.
* `is_nil(message)` is clearer than `message != nil`
* Describing simple closure first is more readable.
```
if is_nil(message) do
  %State{state | build_state: next_build_state}
else
  log = {Time.now(), :info, ContextId.system_context(), message}
  next_log_state = LogRotation.write_log(log_state, log)
  %State{state | log_state: next_log_state, build_state: next_build_state}
end
```

Reply:
Thank you!
I fixed it.
https://github.com/access-company/antikythera/pull/93/commits/b932854ea60a2f373de2f68a48e44aecc0de3546